components/user-mgt/org.wso2.carbon.identity.user.pre.update.password.action/src/main/java/org/wso2/carbon/identity/user/pre/update/password/action/internal/execution/PreUpdatePasswordRequestBuilder.java

```diff
@@ -175,17 +184,22 @@ private PreUpdatePasswordEvent.Action getAction() throws ActionExecutionRequestB
         throw new ActionExecutionRequestBuilderException("Invalid action flow.");
     }
 
-    private static Tenant getTenant() {
+    private static Tenant getTenant(String organizationId) throws ActionExecutionRequestBuilderException {
+
+        if (organizationId != null) {
+            return OrganizationMgtUtil.resolveTenant(organizationId);
+        }
 
         return new Tenant(String.valueOf(IdentityContext.getThreadLocalCarbonContext().getTenantId()),
                 IdentityContext.getThreadLocalCarbonContext().getTenantDomain());
     }
 
-    private User getUser(UserActionContext userActionContext, PreUpdatePasswordAction preUpdatePasswordAction)
-            throws ActionExecutionRequestBuilderException {
+    private User getUser(UserActionContext userActionContext, PreUpdatePasswordAction preUpdatePasswordAction,
+                         Organization organization) throws ActionExecutionRequestBuilderException {
 
         PasswordUpdatingUser.Builder userBuilder = new PasswordUpdatingUser.Builder()
-                .id(userActionContext.getUserActionRequestDTO().getUserId());
+                .id(userActionContext.getUserActionRequestDTO().getUserId())
+                .organization(organization);
```

Comment:
As of now, a user gets managed within the particular organization boundary only. If this is the case this logic is fine.
Is this always true ?